lib/puppet/provider/package/puppet_gem.rb

```diff
@@ -6,20 +6,7 @@
 
   confine :true => Facter.value(:aio_agent_version)
```

Comment:
I debated this, but if we implement gem_cmd in the runmode then is there still a need to confine this to AIO?

Reply:
I'd like to keep this confined to the AIO Ruby, otherwise it will serve the same purpose as the regular `gem` provider.

Having `gem_cmd` as part of the run mode is a good idea though, as it keeps things uncluttered in the provider.

Reply:
My reasoning was that as a user I can then simply use `puppet_gem` to ensure a gem is installed in the Puppet environment. However, it can also lead to conflicts with the regular `gem` provider. So I can see the argument for either. We can also first clean up the platform independence and decide on it later while keeping the status quo.